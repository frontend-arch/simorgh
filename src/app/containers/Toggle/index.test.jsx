import React from 'react';
import { shouldMatchSnapshot, isNull } from '@bbc/psammead-test-helpers';
import ToggleContainer from '.';

jest.mock('react', () => {
  const original = jest.requireActual('react');
  return {
    ...original,
    useContext: jest.fn(),
  };
});

const WithContext = (toggleName, FallbackComponent) => (
  <ToggleContainer
    toggleName={toggleName}
    FallbackComponent={FallbackComponent}
  >
    <h1>Child component</h1>
  </ToggleContainer>
);

const FallbackComponent = () => <p>Fallback UI</p>;

const { useContext } = jest.requireMock('react');

describe('ToggleContainer', () => {
  beforeEach(() => {
    useContext.mockReturnValueOnce({ env: 'test' }).mockReturnValueOnce({
      toggleState: { test: { foo: { enabled: true } } },
    });
  });

  describe('should render child component', () => {
    shouldMatchSnapshot('should match snapshot', WithContext('foo'));
  });

  describe('should render null', () => {
    isNull('should match snapshot', WithContext('bar'));
  });

  describe('should render fallback component', () => {
    shouldMatchSnapshot(
      'should match snapshot',
      WithContext('bar', FallbackComponent),
    );
  });
});
