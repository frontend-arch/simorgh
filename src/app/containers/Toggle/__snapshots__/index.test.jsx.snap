// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleContainer should render child component should match snapshot 1`] = `
<h1>
  Child component
</h1>
`;

exports[`ToggleContainer should render fallback component should match snapshot 1`] = `
<p>
  Fallback UI
</p>
`;
